API no more working.

http://www.mix-it.fr/article/40/wanna-hack-mix-it-

Any non empty JSON response lead to crash...
Since migration to CloudFoundry?

```
2015-03-09T00:24:31.71+0100 [App/0]   OUT [CONTAINER] com.google.gson.ParameterizedTypeHandlerMap        WARNING Overriding the existing type handler for class java.lang.Object
2015-03-09T00:24:31.71+0100 [App/0]   OUT [CONTAINER] com.google.gson.ParameterizedTypeHandlerMap        WARNING Overriding the existing type handler for class java.lang.Object
2015-03-09T00:24:31.76+0100 [App/0]   OUT 23:24:31,760 ERROR ~ 
2015-03-09T00:24:31.76+0100 [App/0]   OUT @6ldghbbhg
2015-03-09T00:24:31.76+0100 [App/0]   OUT Internal Server Error (500)
2015-03-09T00:24:31.76+0100 [App/0]   OUT Execution exception (In /app/controllers/api/JsonpController.java around line 35)
2015-03-09T00:24:31.76+0100 [App/0]   OUT IllegalStateException occured : circular reference error   Offending field: links    Offending object: preserveType: true, type: class models.Member, obj: Olivier Girardot
2015-03-09T00:24:31.76+0100 [App/0]   OUT play.exceptions.JavaExecutionException: circular reference error
2015-03-09T00:24:31.76+0100 [App/0]   OUT   Offending field: links
2015-03-09T00:24:31.76+0100 [App/0]   OUT   Offending object: preserveType: true, type: class models.Member, obj: Olivier Girardot
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:231)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.server.ServletWrapper$ServletInvocation.execute(ServletWrapper.java:557)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.Invoker$Invocation.run(Invoker.java:278)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.server.ServletWrapper$ServletInvocation.run(ServletWrapper.java:548)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.Invoker.invokeInThread(Invoker.java:68)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.server.ServletWrapper.service(ServletWrapper.java:142)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:723)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:622)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:652)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:293)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:861)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:606)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:489)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at java.lang.Thread.run(Thread.java:745)
2015-03-09T00:24:31.76+0100 [App/0]   OUT Caused by: java.lang.IllegalStateException: circular reference error
2015-03-09T00:24:31.76+0100 [App/0]   OUT   Offending field: links
2015-03-09T00:24:31.76+0100 [App/0]   OUT   Offending object: preserveType: true, type: class models.Member, obj: Olivier Girardot
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.CircularReferenceException.createDetailedException(CircularReferenceException.java:43)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitFieldUsingCustomHandler(JsonSerializationVisitor.java:213)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ReflectingFieldNavigator.visitFieldsReflectively(ReflectingFieldNavigator.java:63)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:120)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:624)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.findAndInvokeCustomSerializer(JsonSerializationVisitor.java:184)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitFieldUsingCustomHandler(JsonSerializationVisitor.java:204)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ReflectingFieldNavigator.visitFieldsReflectively(ReflectingFieldNavigator.java:63)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:120)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:624)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.findAndInvokeCustomSerializer(JsonSerializationVisitor.java:184)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitFieldUsingCustomHandler(JsonSerializationVisitor.java:204)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ReflectingFieldNavigator.visitFieldsReflectively(ReflectingFieldNavigator.java:63)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:120)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:624)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.findAndInvokeCustomSerializer(JsonSerializationVisitor.java:184)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitFieldUsingCustomHandler(JsonSerializationVisitor.java:204)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ReflectingFieldNavigator.visitFieldsReflectively(ReflectingFieldNavigator.java:63)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:120)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:624)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.findAndInvokeCustomSerializer(JsonSerializationVisitor.java:184)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitFieldUsingCustomHandler(JsonSerializationVisitor.java:204)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ReflectingFieldNavigator.visitFieldsReflectively(ReflectingFieldNavigator.java:63)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:120)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:624)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.findAndInvokeCustomSerializer(JsonSerializationVisitor.java:184)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitFieldUsingCustomHandler(JsonSerializationVisitor.java:204)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ReflectingFieldNavigator.visitFieldsReflectively(ReflectingFieldNavigator.java:63)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:120)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:624)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.findAndInvokeCustomSerializer(JsonSerializationVisitor.java:184)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitUsingCustomHandler(JsonSerializationVisitor.java:160)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:101)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.Gson.toJsonTree(Gson.java:220)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.Gson.toJson(Gson.java:260)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.Gson.toJson(Gson.java:240)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.results.RenderJson.<init>(RenderJson.java:33)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.Controller.renderJSON(Controller.java:346)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at controllers.api.JsonpController.renderJSON(JsonpController.java:35)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at controllers.api.ApiMembers.sponsors(ApiMembers.java:38)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	... 20 more
2015-03-09T00:24:31.76+0100 [App/0]   OUT Caused by: com.google.gson.CircularReferenceException: circular reference error
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.start(JsonSerializationVisitor.java:61)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:109)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:62)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationContextDefault.serialize(JsonSerializationContextDefault.java:53)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:643)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.DefaultTypeAdapters$CollectionTypeAdapter.serialize(DefaultTypeAdapters.java:624)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.findAndInvokeCustomSerializer(JsonSerializationVisitor.java:184)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	at com.google.gson.JsonSerializationVisitor.visitFieldUsingCustomHandler(JsonSerializationVisitor.java:204)
2015-03-09T00:24:31.76+0100 [App/0]   OUT 	... 83 more


```

--- a/app/controllers/api/AbstractMemberJsonSerializer.java
+++ b/app/controllers/api/AbstractMemberJsonSerializer.java
@@ -30,12 +30,6 @@ public abstract class AbstractMemberJsonSerializer {
         result.addProperty("urlimage", member.getUrlImage());
         result.addProperty("nbConsults", member.nbConsults);
 
-        if (CollectionUtils.size(member.links) != 0) {
-            result.add("links", JSON.toJsonArrayOfIds(member.links));
-        }
-        if (CollectionUtils.size(member.linkers) != 0) {
-            result.add("linkers", JSON.toJsonArrayOfIds(member.linkers));
-        }
         if (CollectionUtils.size(member.sharedLinks) != 0) {
             result.add("sharedLinks", jsonSerializationContext.serialize(member.sharedLinks));
         }

